client/components/backup-retention-management/index.tsx

```diff
@@ -0,0 +1,239 @@
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_7
+		),
+		[ RETENTION_OPTIONS.RETENTION_DAYS_30 ]: usePrepareRetentionOptions(
+			translate( '30 days' ),
+			RETENTION_OPTIONS.RETENTION_DAYS_30,
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_30
+		),
+		[ RETENTION_OPTIONS.RETENTION_DAYS_120 ]: usePrepareRetentionOptions(
+			translate( '120 days' ),
+			RETENTION_OPTIONS.RETENTION_DAYS_120,
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_120
+		),
+		[ RETENTION_OPTIONS.RETENTION_DAYS_365 ]: usePrepareRetentionOptions(
+			translate( '1 year' ),
+			RETENTION_OPTIONS.RETENTION_DAYS_365,
+			currentRetentionPlan,
+			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_365
+		),
+	};
+
+	const updateRetentionRequestStatus = useSelector( ( state ) =>
+		getBackupRetentionUpdateRequestStatus( state, siteId )
+	);
+
+	// Set the retention period selected when the user selects a new option
+	const onRetentionSelectionChange = useCallback( ( value: number ) => {
+		if ( value ) {
+			setRetentionSelected( value );
+		}
+	}, [] );
+
+	const disableFormSubmission =
+		! retentionSelected ||
+		retentionSelected === currentRetentionPlan ||
+		retentionOptionsCards[ retentionSelected ].upgradeRequired ||
+		updateRetentionRequestStatus === BACKUP_RETENTION_UPDATE_REQUEST.PENDING;
+
+	const [ confirmationDialogVisible, setConfirmationDialogVisible ] = useState( false );
+	const onConfirmationClose = useCallback( () => {
+		setConfirmationDialogVisible( false );
+	}, [] );
+
+	const updateRetentionPeriod = useCallback( () => {
+		dispatch( updateBackupRetention( siteId, retentionSelected as RetentionPeriod ) );
+	}, [ dispatch, retentionSelected, siteId ] );
+
+	const handleUpdateRetention = useCallback( () => {
+		if ( ! retentionSelected ) {
+			return;
+		}
+
+		// Show confirmation dialog if updating retention period to a lower value, otherwise update immediately
+		if ( retentionSelected < currentRetentionPlan ) {
+			setConfirmationDialogVisible( true );
+		} else {
+			updateRetentionPeriod();
+		}
+	}, [ currentRetentionPlan, retentionSelected, updateRetentionPeriod ] );
+
+	// Set the retention period selected when we fetch the current plan retention period
+	useEffect( () => {
+		if ( currentRetentionPlan && ! retentionSelected ) {
+			setRetentionSelected( currentRetentionPlan );
+		}
+	}, [ currentRetentionPlan, retentionSelected ] );
+
+	useEffect( () => {
+		if ( updateRetentionRequestStatus === BACKUP_RETENTION_UPDATE_REQUEST.SUCCESS ) {
+			// Update the current retention plan to the one updated by the user.
+			setCurrentRetentionPlan( retentionSelected );
+			setConfirmationDialogVisible( false );
+		} else if ( updateRetentionRequestStatus === BACKUP_RETENTION_UPDATE_REQUEST.FAILED ) {
+			setConfirmationDialogVisible( false );
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why is this necessary?

Reply:
@tyxla in this case, I would like this check to occur only when `updateRetentionRequestStatus` changes. If I don't include this line, the linter will keep expecting to pass all as dependencies.

Reply:
Technically, the rest of the dependencies can also change and that theoretically should re-run the effect. Is there a good reason to seek workarounds for this default behavior?

Reply:
I think I see now what's the issue and how it should be.

After updating the retention period, I'm setting the `retentionSelected` at that moment as the `currentRetentionPlan`, which is not the best approach. The right way should be to update the retentionPlan on the redux state using an action dispatcher in the [onSuccess callback](https://github.com/Automattic/wp-calypso/pull/73035/files#diff-d9ef76a91dcc2d8e7451ccddfbc3fc6b732bba3c09dc25de5f55da431b361c2fR24).

I'll take a double look and handle it in a new PR.

Reply:
Just [submitted a PR](https://github.com/Automattic/wp-calypso/pull/73300) for review related to this, it would be great if you can take a look :) https://github.com/Automattic/wp-calypso/pull/73300